src/components/ProtectedRoute.jsx

```diff
@@ -0,0 +1,18 @@
+import { Navigate, useLocation } from 'react-router-dom';
+import { useAuthContext } from '@/context/AuthContext';
+
+const ProtectedRoute = ({ children }) => {
```

Comment:
- [ ] You did a good job working on your ProtectedRoute.jsx 💯 However, please make sure that the {addTodoItem} has the prop validation  kindly fix it. 🤓 🔧